1_fetch/src/find_nwis_sites.R

```diff
@@ -0,0 +1,19 @@
+get_drb_sites <- function(hucs,pcodes_select,site_tp_select){
+  
+  # hucs is a character vector containing the minor HUCs (hydrologic unit codes) over which to search. A minor HUC is 8 digits in length. 
+  # pcodes_select is a character vector containing the USGS parameter codes of interest
+  # site_tp_select is a character vector containing the USGS site types to retain
+  
+  # Search for sites separately by HUC8 region since calls to readNWISdata allow no more than 10 minor HUCs to be specified:
+  drb_sites_ls <- lapply(hucs,function(x)
+    readNWISdata(huc=x,parameterCd=pcodes_select,service="site",seriesCatalogOutput=TRUE))
```

Comment:
I think you can include siteType in the service call which would let you filter upfront instead of with the lines below. https://waterservices.usgs.gov/rest/Site-Service.html#siteType

Reply:
I actually didn't know this, so that's a great tip! However, I noticed this sentence from the documentation:

> If you request a major site type (ex: &siteType=ST) you will get all sub-site types of the same major type as well (in this case, ST-CA, ST-DCH and ST-TS)

We request stream samples ("ST") but we don't want tidal samples for this work ("ST-TS"). Adding "ST" to siteType in the service call would then still require a filtering step to remove tidal sites, so I've decided to keep this code chunk as is and filter for site types within the dataRetrieval output.